07_errors/n0npax/errors.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"fmt"
+)
+
+// Error codes
+const (
+	ErrInvalidInputCode  = 422
+	ErrNotFoundCode      = 404
+	ErrInternalErrorCode = 500
+)
+
+// Error descriptions
+const (
+	PuppyNotFoundMsg   = "Puppy with ID %v not found"
+	InvalidInputMsg    = "Puppy value have to be positive number"
+	DataDecodeErrorMsg = "Internal error. Could not cast stored data to puppy object"
+	CorruptedIDMsg     = "ID is corrupted. Please ensure object ID matched provided ID"
+)
+
+// Error wrapps errors with code, message and error itself
+type Error struct {
+	Message string
+	Code    int
+}
+
+// Error returns error as a string
+func (e *Error) Error() string {
+	return fmt.Sprintf("Error code: %v, message : %v", e.Code, e.Message)
+}
+
+// Errorf creates a new Error with formatting
+func Errorf(code int, msg string) *Error {
```

Comment:
Usually the trailing `f` as in `Printf`, `Equalf` etc. stands for `format` and uses a variadiac function signature à la printf, at least as the latter part of the function signature: fmt.Printf(format, args...) → fmt.Printf(format, arg1, arg2, ....) →fmt.Printf("name %s, age: %d\d", name, age) ...

Reply:
Fixed.

Anyway, I was using const with error message to avoid running `sed` on source code.
Whole refactor looks like (I know I can run single sed to clean everything).
```
0127  sed -i 's/total/nextID/g' **/*.go
10128  sed -i 's/ErrInvalidInput(/Errorf(puppy.ErrInvalidInputCode, /g' **/*.go                                                 
10129  sed -i 's/ErrInvalidInput(/Errorf(puppy.InvalidInputMsg, /g' **/*.go                                                     
10130  sed -i 's/ErrInvalidInput(/Errorf(puppy.InvalidInputMsg/g' **/*.go                                                       
10137  sed -i 's/ErrInternalError(/Errorf(puppy.ErrInternalErrorCode, /g' **/*.go                                               
10148  sed -i 's/PuppyNotFoundMsg/"Puppy with ID (%v) not found"/g'  *.go                                                       
10150  sed -i 's/InvalidInputMsg/"Puppy value have to be positive number"/g' *.go                                               
10151  sed -i 's/DataDecodeErrorMsg/"Internal error. Could not cast stored data to puppy object"/g' *.go                        
10152  sed -i 's/CorruptedIDMsg/"ID is corrupted. Please ensure object ID matched provided ID"/g'  *.go                         
10156  sed -i 's/ErrNotFound(/Errorf(puppy.ErrNotFoundCode, /g'  *.go                                                           
10158  sed -i 's/puppy\.Err/Err/g' *.go
```